src/components/shopping-cart/shopping-card.tsx

```diff
@@ -0,0 +1,122 @@
+import images from 'assets';
+import { Link } from 'expo-router';
+import { memo } from 'react';
+
+import { usePatchShoppingCartItem } from '@/api/cart/use-patch-line-item';
+import { Image, showErrorMessage, Text, TouchableOpacity, View } from '@/ui';
+
+export interface ShoppingProductCardProps {
+  id: number;
+  name: string;
+  price: string;
+  state: string;
+  image_url: string;
+  quantity: number;
+  refetch: () => void;
+}
+
+type QuantityControllerProps = {
+  quantity: number;
+  handlePress: (newQuantity: number) => void;
+};
+
+const QuantityController = ({
+  quantity,
+  handlePress,
+}: QuantityControllerProps) => {
+  return (
+    <View className="flex-row items-center gap-4">
+      <TouchableOpacity
+        onPress={() => {
+          handlePress(0);
+        }}
+      >
+        <Image
+          source={images.trashIcon()}
+          contentFit="contain"
+          className="h-4 w-4"
+        />
+      </TouchableOpacity>
+      <Text>{quantity}</Text>
+      <TouchableOpacity
+        onPress={() => {
+          handlePress(quantity + 1);
+        }}
+      >
+        <Image
+          source={images.plusIcon()}
+          contentFit="contain"
+          className="h-4 w-4"
+        />
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+export const ShoppingCard = memo(
+  ({
+    name,
+    price,
+    state,
+    image_url,
+    id,
+    quantity,
+    refetch,
+  }: ShoppingProductCardProps) => {
+    const { mutate: patchShoppingCartItem } = usePatchShoppingCartItem({});
+
+    const handlePatch = (newQuantity: number) => {
+      try {
+        patchShoppingCartItem({ itemId: id, quantity: newQuantity });
```

Comment:
It seems like an await is missing here, refetching without patchShoppingCartItem being finished would not be helpful

Reply:
An alternative to handling this would be to use `onSuccess` and `onError` callbacks.

`usePatchShoppingCartItem` implements `useBaseMutations`, instead of passing an empty object you can pass both callbacks.

Try/catch statement can be deleted if you use the above approach, it would be my primary suggestion to handle success/error cases within a mutation.